docs/src/main/asciidoc/tls-registry-reference.adoc

```diff
@@ -0,0 +1,518 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
+////
+= TLS registry reference
+include::_attributes.adoc[]
+:categories: network
+:summary: TLS registry configuration and usage
+:numbered:
+:sectnums:
+:sectnumlevels: 4
+:topics: TLS, http, SSL, HTTPS, security, network
+:extensions: io.quarkus:quarkus-tls-registry
+
+The TLS registry is a Quarkus extension centralizing the TLS configuration for the application.
+It allows to define the TLS configuration in a single place and to reference it from multiple places in the application.
+
+The TLS extension should be automatically added to your project as soon as you use a compatible extension.
+For example, if your application uses Quarkus REST, gRPC or reactive routes, the TLS registry is automatically added to your project.
+
+== Examples
+
+So configure a TLS connection, and more especially the key stores and trust stored, you use the `quarkus.tls.*` properties.
+
+Configuration directly under `quarkus.tls` is the default configuration that will be used by all the TLS connections in the application.
+However, you can also have specific configurations for specific connections by using the `quarkus.tls.<name>.*` properties.
+
+=== Configure the HTTP server to use https://
+
+To configure the HTTP server to use HTTPS, you can use the following configuration:
+
+[source,properties]
+----
+quarkus.tls.key-store.pem[0].cert=server.crt
+quarkus.tls.key-store.pem[0].key=server.key
```

Comment:
Why would a map offer? There is no "name" in this case.

Reply:
We've generally tried to avoid this array stuff as much as possible as it's really weird and our tooling can't help much with it. 

The name can be arbitrary (as it is with all other named confifgurations)

Reply:
Ok, let me try this.

Reply:
As the PEM pair order must be preserved, Maps can be problematic.
See the full discussion:
https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/Order.20of.20Map.20in.20config.3F